### PR TITLE
Rotative SVG

### PR DESCRIPTION
**Pull request by arnauddieumegard**
Add support for SVG images in Rotative WorkspaceImage.
Add RCPTT tests fixes
Rename and move rotative image plugins

### CODE REVIEW

File: common/plugins/org.polarsys.kitalpha.common.sirius.extension.rotativeimage/src/org/polarsys/kitalpha/common/sirius/extension/rotativeimage/editpart/RotativeImageEditPartProvider.java [hunk 1]
```diff
@@ -34,26 +35,42 @@
...
+			    if (rotativeDescription != null) {
+		            return true;
+		        }
+			}
+		}
+		return false;
+	}
+
+	private boolean isRotativeImageEditPart(final EObject resolvedSemanticElement) {
+		final WorkspaceImage customStyle = (WorkspaceImage) resolvedSemanticElement;
```
[the remark this answers is not in the document]
Comment (arnauddieumegard) [reply common/plugins/org.polarsys.kitalpha.common.sirius.extension.rotativeimage/src/org/polarsys/kitalpha/common/sirius/extension/rotativeimage/editpart/RotativeImageEditPartProvider.java]: Yes

File: common/plugins/org.polarsys.kitalpha.common.sirius.extension.rotativeimage/src/org/polarsys/kitalpha/common/sirius/extension/rotativeimage/editpart/RotativeImageEditPartProvider.java [hunk 2]
```diff
@@ -34,26 +35,42 @@
...
-					}
+				if (isRotativeDNode4editPart(resolvedSemanticElement)) {
+					return RotativeDNode4EditPart.class;
 				}
 			}
 		}
 		return super.getNodeEditPartClass(view);
 	}
 
+	private boolean isRotativeDNode4editPart(final EObject resolvedSemanticElement) {
```
[the remark this answers is not in the document]
Comment (arnauddieumegard) [reply common/plugins/org.polarsys.kitalpha.common.sirius.extension.rotativeimage/src/org/polarsys/kitalpha/common/sirius/extension/rotativeimage/editpart/RotativeImageEditPartProvider.java]: Yes, will do

File: common/plugins/org.polarsys.kitalpha.common.sirius.extension.rotativeimage/src/org/polarsys/kitalpha/common/sirius/extension/rotativeimage/internal/helpers/RotativeWorkspaceImageHelper.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+		case PositionConstants.WEST:
+		case PositionConstants.EAST:
+			return rect.transpose();
+		default:
+			return rect;
+		}
+	}
+
+	public static String get4ImagesDocumentKey(String path, int orientation) {
+		int extentionPosition = path.lastIndexOf(".");
```
[the remark this answers is not in the document]
Comment (arnauddieumegard) [reply common/plugins/org.polarsys.kitalpha.common.sirius.extension.rotativeimage/src/org/polarsys/kitalpha/common/sirius/extension/rotativeimage/internal/helpers/RotativeWorkspaceImageHelper.java]: Nice catch.

File: common/plugins/org.polarsys.kitalpha.common.sirius.extension.rotativeimage/src/org/polarsys/kitalpha/common/sirius/extension/rotativeimage/Activator.java [hunk 4]
```diff
@@ -33,18 +33,33 @@
...
+	    }
 		return descriptions;
 	}
 	
 	public RotativeDescription getBestDescription(String id) {
-		if (id==null) return null;
-		for (RotativeDescription desc : descriptions) {
-			if (id.startsWith(desc.id)) return desc;
+	    if (id==null) return null;
+		if (descriptions == null) {
```
[the remark this answers is not in the document]
Comment (arnauddieumegard) [reply common/plugins/org.polarsys.kitalpha.common.sirius.extension.rotativeimage/src/org/polarsys/kitalpha/common/sirius/extension/rotativeimage/Activator.java]: You are right

**Review by pdulth — APPROVED** [review]

**Comment by pdulth** [comment]
@bzendagui wdyt?

File: sirius/tests/plugins/org.polarsys.kitalpha.sirius.rotativeimage.junit.tests/OSGI-INF/l10n/bundle.properties [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+ #  http://www.eclipse.org/legal/epl-2.0
+ #
+ #  SPDX-License-Identifier: EPL-2.0
+ # 
+ # Contributors:
+ #  Thales Global Services S.A.S - initial API and implementation
+ ##############################################################################
+ 
+ ##
+ # @author Boubekeur Zendagui
```
Comment (bzendagui) [inline sirius/tests/plugins/org.polarsys.kitalpha.sirius.rotativeimage.junit.tests/OSGI-INF/l10n/bundle.properties:14]: ???

**Review by bzendagui — APPROVED** [review]
It good for me except an author name in one .properties file !! :)

File: sirius/tests/plugins/org.polarsys.kitalpha.sirius.rotativeimage.junit.tests/OSGI-INF/l10n/bundle.properties (hunk 5 above)
Comment (arnauddieumegard) [reply sirius/tests/plugins/org.polarsys.kitalpha.sirius.rotativeimage.junit.tests/OSGI-INF/l10n/bundle.properties:14]: Damn ! Thanks for noticing

**Pull request merged**